### PR TITLE
UI updates to GCP-projects plugin

### PR DESCRIPTION
**Pull request by jeremyguarini**
Adds the following to the project list page:
* pagination
* filtering
* sorting
* rows per page
* show/hide columns

Makes breadcrumb a link back to project list for the project details and new project views.

In project list page, updates New project button to use RouterLink instead of href to avoid login prompt.

In project details view, links to project details and logs now work, clicking on these will open the project or logs in GCP in new tab.

Signed-off-by: Jeremy Guarini <jguarini@paloaltonetworks.com>

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by jeremyguarini** [comment]
some screen shots showing the updates
Screen Shot 2021-10-29 at 9 49 51 AM

Screen Shot 2021-10-29 at 9 50 02 AM

Screen Shot 2021-10-29 at 9 50 34 AM

Screen Shot 2021-10-29 at 9 50 22 AM

**Review by Rugvip — COMMENTED** [review]
Thank you! 👍 

Just a concern about ending up with too many different tables. The current one is used all over the place, here's one place as an example: https://github.com/backstage/backstage/blob/a4b86485a3aac3fa511bb349cdbbea09c0e42fd5/plugins/rollbar/src/components/RollbarTopItemsTable/RollbarTopItemsTable.tsx#L118

File: plugins/gcp-projects/package.json [hunk 1]
```diff
@@ -35,6 +35,7 @@
     "@backstage/core-plugin-api": "^0.1.12",
     "@backstage/theme": "^0.2.12",
     "@material-ui/core": "^4.12.2",
+    "@material-ui/data-grid": "^4.0.0-alpha.37",
```
[the remark this answers is not in the document]
Comment (jeremyguarini) [reply plugins/gcp-projects/package.json:38]: Will work on switching to material-table.

[the remark this answers is not in the document]
Comment (jeremyguarini) [reply plugins/gcp-projects/package.json]: Code updated to use material-table. Was a bit easier than I expected.

[the remark this answers is not in the document]
Comment (Rugvip) [reply plugins/gcp-projects/package.json]: A shiet, so we have our own [Table](https://backstage.io/storybook/?path=/story/data-display-table--default-table) component that applies some custom styling, gives you the icons, etc. Sorry if that wasn't more clear 😅

Alright to spend a minimal amount of effort to see if it's possible to switch over? Should remove some of the setup clutter with the icons etc and the API is mostly compatible. If it's not a smooth switch that's fine too and we could potentially merge as is

[the remark this answers is not in the document]
Comment (jeremyguarini) [reply plugins/gcp-projects/package.json]: @Rugvip It was a simple enough of a change to swap-out the material-table for Table. Let me know if there is anything else that should be done.

File: yarn.lock [hunk 2]
```diff
@@ -4416,6 +4416,16 @@
     react-is "^16.8.0 || ^17.0.0"
     react-transition-group "^4.4.0"
 
+"@material-ui/data-grid@^4.0.0-alpha.37":
```
Comment (Rugvip) [inline yarn.lock:4419]: Not sure how this is getting through CI tbh, will have to have a look at that :p

Should revert these changes to `yarn.lock` either way

**Review by Rugvip — COMMENTED** [review]
Spotted one tiny thing, but otherwise good to go! 👍

File: yarn.lock (hunk 2 above)
Comment (jeremyguarini) [reply yarn.lock]: hmm yeh I should have reran yarn install without --frozen-lockfile to get that dep removed.

Comment (jeremyguarini) [reply yarn.lock]: @Rugvip thanks for the reviews.

**Review by Rugvip — APPROVED** [review]
Thank you for the contribution! 👍 

Let's :shipit:

**Pull request merged**